### PR TITLE
docs(peering): remove beta references

### PR DESCRIPTION
**Pull request by DanStough**
### Description
Cluster peering will be GA for the Consul 1.14 release. This PR removes the beta references from the site.

There will be subsequent PRs to handle other content updates, namely:
1. Control Plane usage
1. Mesh gateway control
1. Changes to ACL permissions

### CODE REVIEW

**Comment by hc-github-team-consul-core** [comment]
All unit tests passed! Your changes bring test coverage over affected code to 48.4%. Total coverage in main is 63.8%. [See coverage delta for your changes](https://output.circle-artifacts.com/output/job/2dd46781-897f-4c72-bb9e-ba3798487ae8/artifacts/0/tmp/output/diff_coverage_heatmap.html)

File: website/content/docs/connect/cluster-peering/index.mdx [hunk 1]
```diff
@@ -43,23 +41,11 @@ Regardless of whether you connect your clusters through WAN federation or cluste
...
-- You can use the [`peering` rule for ACL enforcement](/docs/security/acl/acl-rules#peering) of peering APIs.
+## Important Cluster Peering Constraints
 
-Not all features and functionality are available in the beta release. In particular, consider the following technical constraints:
+Consider the following technical constraints:
 
-- Mesh gateways for _server to server traffic_ are not available.
 - Services with node, instance, and check definitions totaling more than 50MB cannot be exported to a peer.
-- The `service-splitter` and `service-router` configuration entry kinds cannot directly target a peer. To split or route traffic to a service instance on a peer, you must supplement your desired dynamic routing definition with a `service-resolver` config entry on the dialing cluster. Refer to [L7 traffic management between peers](/docs/connect/cluster-peering/create-manage-peering#L7-traffic) for more information.
+- Two admin partitions in the same datacenter cannot be peered. Consider using [`exported-services`](/docs/connect/config-entries/exported-services#exporting-services-to-peered-clusters) directly.
```
Comment (erichaberkorn) [inline website/content/docs/connect/cluster-peering/index.mdx:49]:
```suggestion
- Two admin partitions in the same datacenter cannot be peered. Use [`exported-services`](/docs/connect/config-entries/exported-services#exporting-services-to-peered-clusters) directly.
```

**Review by erichaberkorn — COMMENTED** [review]
Did you mean to remove .github/go_test_coverage.txt?

**Review by erichaberkorn — APPROVED** [review]
It looks like the test coverage file is a bot. This looks good to me 👍

**Review by hashi-derek — APPROVED** [review]

**Pull request merged**